### ISSUE TITLE
Add cluster config option support

### ISSUE THREAD

**Issue by leggetter** [pusher/pusher-http-php#70]
A cluster ID (`eu`/`us-east2`) and the URL is formed for the developer. This is how things work across all recently libraries.

```php
$options = array('cluster' => 'eu');
$pusher = new Pusher('app-id', 'app-key', 'app-secret', $options);
```

The HTTP API (`$baseUrl`) will be:

```
https://api-eu.pusher.com
```

If `` were provided as the cluster ID then the URL would be:

```
https://api-useast2.pusher.com
```

**Comment by tanuck** [comment]
I think the node.js library already has this? There's no verification of cluster names though, so in theory they will get a 404 upon the api request if they pass a bad name to the library constructor!

Do you want to go down the same line here, or would you like to verify they've chosen a valid cluster?

**Comment by leggetter** [comment]
Although I like the idea of validating the cluster this would probably require a pre-defined list of valid clusters. Obviously we can add clusters so every time we create a new cluster it would then require an update to the library. So we'd best avoid this and hope the 404 is a good enough indicator that the Cluster ID is incorrect.

**Comment by tanuck** [comment]
@leggetter this can be closed now.

**Comment by leggetter** [comment]
@tanuck :+1:

Fixed by #73.
